src/plugins/data/public/ui/query_editor/language_selector.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiComboBox, EuiComboBoxOptionOption, PopoverAnchorPosition } from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import React from 'react';
+import { getUiService } from '../../services';
+
+interface Props {
+  language: string;
+  onSelectLanguage: (newLanguage: string) => void;
+  anchorPosition?: PopoverAnchorPosition;
+  appName?: string;
+}
+
+const mapExternalLanguageToOptions = (language: string) => {
+  return {
+    label: language,
+    value: language,
+  };
+};
+
+export const QueryLanguageSelector = (props: Props) => {
+  const dqlLabel = i18n.translate('data.query.queryEditor.dqlLanguageName', {
+    defaultMessage: 'DQL',
+  });
+  const luceneLabel = i18n.translate('data.query.queryEditor.luceneLanguageName', {
+    defaultMessage: 'Lucene',
+  });
+
+  const languageOptions: EuiComboBoxOptionOption[] = [
+    {
+      label: dqlLabel,
+      value: 'kuery',
+    },
+    {
+      label: luceneLabel,
+      value: 'lucene',
+    },
+  ];
+
+  const uiService = getUiService();
+
+  const queryEnhancements = uiService.Settings.getAllQueryEnhancements();
+  queryEnhancements.forEach((enhancement) => {
+    if (
+      (enhancement.supportedAppNames &&
+        props.appName &&
+        !enhancement.supportedAppNames.includes(props.appName)) ||
+      uiService.Settings.getUserQueryLanguageBlocklist().includes(
+        enhancement.language.toLowerCase()
+      )
+    )
+      return;
+    languageOptions.unshift(mapExternalLanguageToOptions(enhancement.language));
```

Comment:
related to the issue with single line if conditions. so will track this on that issue to clean it up for readability.

basically i skip adding a language option if the current app name is not a supported app name. also skip it if the author of the search interceptor decided they don't want to support specific apps.

so basically on the OpenSearch Dashboards we can decide to only show additional languages on specific appNames (right now `discover`) and then I can make sure PPL is only added to Discover.

the handling of these both cases might be unneeded now since we are splittling out the components differently. but i originally implemented this in mind with the ability to add this to any part of the application besides Discover.

Reply:
loll this is so tricky cuz return does seem to skip unshift line. now I understand from the explain: the return statement skips the unshift for enhancements that are either not supported by the current app or are blocked. maybe given a name to the if clause can add readability.